R/mainTests.r

```diff
@@ -1153,6 +1160,454 @@ country_code <- function(data) {
+                  "result" = "FAIL"
+                )
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  return(output)
+}
+
+
+# other_geography_duplicates  ----------------------------------------
+# check that there is a 1:1 relationship between geography codes and names
+# ignore school
+lower_level_geog_names <- geography_matrix[c(7:12, 14:16), 2:3] %>% as.character()
+
+other_geography_duplicates <- function(data) {
+  if (!any(lower_level_geog_names %in% names(data))) {
+    output <- list(
+      "message" = "Lower-level geography data is not present in this data file.",
+      "result" = "IGNORE"
+    )
+  } else {
+    geog_data <- data %>%
+      select(any_of(c(
+        "geographic_level", lower_level_geog_names
+      ))) %>%
+      distinct() %>%
+      mutate(ID = 1:n())
+
+
+    names <- geog_data %>%
+      select(ID, geographic_level, contains("name")) %>%
+      mutate_if(is.numeric, as.character) %>%
+      melt(id.vars = c("ID", "geographic_level"), na.rm = TRUE) %>%
+      select(ID, geographic_level, name = value)
+
+    codes <- geog_data %>%
+      select(ID, geographic_level, !contains("name")) %>%
+      mutate_if(is.numeric, as.character) %>%
+      melt(id.vars = c("ID", "geographic_level"), na.rm = TRUE) %>%
+      select(ID, geographic_level, code = value)
+
+    lookup_creator <- names %>%
+      full_join(codes, by = c("ID", "geographic_level")) %>%
+      select(-ID, -geographic_level) %>%
+      distinct() %>%
+      add_count(name, name = "name_n") %>%
+      add_count(code, name = "code_n")
+
+    multi_count_name <- lookup_creator %>%
+      filter(name_n > 1) %>%
+      mutate(combo = paste0(name, " - ", name_n, " different codes")) %>%
+      select(combo) %>%
+      distinct() %>%
+      pull()
+
+
+    if (length(multi_count_name) == 0) {
+      output <- list(
+        "message" = "every geography has one code.",
+        "result" = "PASS"
+      )
+    } else {
+      if (length(multi_count_name) == 1) {
+        output <- list(
+          "message" = paste0("The following geography has multiple codes:'", paste0(multi_count_name), "'.
+                             <br> - Each geography should have only one code."),
+          "result" = "FAIL"
+        )
+      } else {
+        if (length(multi_count_name) > 1) {
+          output <- list(
+            "message" = paste0("The following geographies have multiple codes: '", paste0(multi_count_name, collapse = "', '"), "'.
+                             <br> - Each geography should have only one code."),
+            "result" = "FAIL"
+          )
+        }
+      }
+    }
+  }
+  return(output)
+}
+
+# other_geography_code_duplicates  ----------------------------------------
+# check that there is a 1:1 relationship between geography names and codes
+
+lower_level_geog_names <- geography_matrix[c(7:12, 14:16), 2:3] %>% as.character()
+
+other_geography_code_duplicates <- function(data) {
```

Comment:
Is the lookup_creator object the same as the previous test? If so, is it possible to combine this test with that test (other_geography_duplicates)? It would save a little bit of code, though still works fine as separate if it's faffy or not easily done

Reply:
I had it together originally, but the outputs looked quite untidy if there were multiple codes with multiple names/vice versa! Think splitting it out made it easier to decode the fail message